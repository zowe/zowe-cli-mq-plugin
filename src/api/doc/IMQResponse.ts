/*
* This program and the accompanying materials are made available under the terms of the
* Eclipse Public License v2.0 which accompanies this distribution, and is available at
* https://www.eclipse.org/legal/epl-v20.html
*
* SPDX-License-Identifier: EPL-2.0
*
* Copyright Contributors to the Zowe Project.
*
*/


interface ICommandResponses {
    /**
     * The command completion code.
     * @type{string}
     */
    completionCode: string;

    /**
     * The command reason code.
     * @type{string}
     */
    reasonCode: string;

    /**
     * The command response text.
     * @type{string[]}
     */
    text: string[];
}
/**
 * The IMQResponse API response.
 * @export
 */
export interface IMQResponse {

    /**
     * The command response text.
     * @type{string[]}
     */
    commandResponse: ICommandResponses[];

    /**
     * The completion code object.
     * @type{string}
     */
    overallCompletionCode: string;

    /**
     * The reaason code object.
     * @type{string}
     */
    overallReasonCode: string;
}
