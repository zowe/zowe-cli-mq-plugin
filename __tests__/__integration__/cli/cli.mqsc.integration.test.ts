/*
* This program and the accompanying materials are made available under the terms of the
* Eclipse Public License v2.0 which accompanies this distribution, and is available at
* https://www.eclipse.org/legal/epl-v20.html
*
* SPDX-License-Identifier: EPL-2.0
*
* Copyright Contributors to the Zowe Project.
*
*/

// Test environment will be populated in the "beforeAll"
import { ITestEnvironment, TestEnvironment, runCliScript } from "@zowe/cli-test-utils";
import { ITestPropertiesSchema } from "../../__src__/doc/ITestPropertiesSchema";

let TEST_ENVIRONMENT: ITestEnvironment<ITestPropertiesSchema>;
describe("mq mqsc", () => {

    // Create the unique test environment
    beforeAll(async () => {
        TEST_ENVIRONMENT = await TestEnvironment.setUp({
            testName: "mqsc_command",
            installPlugin: true,
            skipProperties: true
        });
    });

    afterAll(async () => {
        await TestEnvironment.cleanUp(TEST_ENVIRONMENT);
    });

    it("should display the mqsc help", async () => {
        const response = await runCliScript(__dirname + "/__scripts__/mqsc_help.sh", TEST_ENVIRONMENT);
        expect(response.stderr.toString()).toBe("");
        expect(response.status).toBe(0);
        expect(response.stdout.toString()).toMatchSnapshot();
    });
});
