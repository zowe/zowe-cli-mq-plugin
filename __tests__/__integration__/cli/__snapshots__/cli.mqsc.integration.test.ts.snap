// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mq mqsc should display the mqsc help 1`] = `
"===============MQ MQSC HELP===============

 COMMAND NAME
 ------------

   mqsc

 DESCRIPTION
 -----------

   MQ Utilities

 USAGE
 -----

   zowe mq run mqsc <qmgr> <cmd> [options]

 POSITIONAL ARGUMENTS
 --------------------

   qmgr		 (string)

      The queue manager to apply the command to

   cmd		 (string)

      The MQSC command

 MQ CONNECTION OPTIONS
 ---------------------

   --host | -H (string)

      The host name used to access the IBM MQ REST API. This might be the host name of
      the IBM MQ mqweb server, or the Zowe API Mediation Layer..

   --port | -P (number)

      The port number used to access the IBM MQ REST API. This might be the port
      number of the IBM MQ mqweb server, or the Zowe API Mediation Layer.

   --user | -u (string)

      The mainframe (MQ) user name, which can be the same as your TSO login.

   --password | --pass | --pw (string)

      The mainframe (MQ) password, which can be the same as your TSO password.

   --reject-unauthorized | --ru (boolean)

      Reject self-signed certificates.

      Default value: false

   --protocol (string)

      Specifies the MQ protocol (http or https).

      Default value: https
      Allowed values: http, https

 PROFILE OPTIONS
 ---------------

   --mq-profile | --mq-p (string)

      The name of a (mq) profile to load for this command execution.

   --base-profile | --base-p (string)

      The name of a (base) profile to load for this command execution.

 BASE CONNECTION OPTIONS
 -----------------------

   --token-type | --tt (string)

      The type of token to get and use for the API. Omit this option to use the
      default token type, which is provided by 'zowe auth login'.

   --token-value | --tv (string)

      The value of the token to pass to the API.

   --cert-file (local file path)

      The file path to a certificate file to use for authentication.

      Note: The CLI does not support certificate files that require a password. For
      more information, search Troubleshooting PEM Certificates in Zowe Docs.

   --cert-key-file (local file path)

      The file path to a certificate key file to use for authentication

 GLOBAL OPTIONS
 --------------

   --show-inputs-only (boolean)

      Show command inputs and do not run the command

   --response-format-json | --rfj (boolean)

      Produce JSON formatted data from a command

   --help | -h (boolean)

      Display help text

   --help-web | --hw (boolean)

      Display HTML help in browser

 EXAMPLES
 --------

   - The following sequence shows how to query a
   server-connection channel that is called NEWSVRCONN on an MQ queue manager - our
   example queue manager is called MQ99:

      $ zowe mq run mqsc MQ99 \\"DISPLAY CHANNEL(NEWSVRCONN)\\"

"
`;
